quickwit/quickwit-serve/src/elasticsearch_api/bulk_v2.rs

```diff
@@ -148,19 +165,69 @@ pub(crate) async fn elastic_bulk_ingest_v2(
                     None,
                 )
             })?;
-        for es_doc_id in es_doc_ids {
+        let index_id = success
+            .index_uid
+            .map(|index_uid| index_uid.index_id)
+            .expect("`index_uid` should be a required field");
+
+        // Populate the response items with one error per parse failure.
+        for parse_failure in success.parse_failures {
+            // Since the generated doc UIDs are monotonically increasing, and inserted in order, we
+            // can find doc handles using binary search.
+            assert!(
+                doc_handles
+                    .windows(2)
+                    .all(|doc_handles| doc_handles[0].doc_uid < doc_handles[1].doc_uid),
+                "doc handles should be sorted by doc UID"
+            );
+            let doc_handle_idx = doc_handles
+                .binary_search_by(|doc_handle| doc_handle.doc_uid.cmp(&parse_failure.doc_uid()))
```

Comment:
nitpick
I think with `Copy` stuff you can use binary_search_by_key`.

Reply:
good point.